src/Features/Core/Portable/NavigationBar/IRemoteNavigationBarItemService.cs

```diff
@@ -44,29 +44,23 @@ internal sealed class SerializableNavigationBarItem
         [DataMember(Order = 8)]
         public readonly bool IsObsolete;
         [DataMember(Order = 9)]
-        public readonly ImmutableArray<TextSpan> Spans;
-        [DataMember(Order = 10)]
-        public readonly TextSpan? SelectionSpan;
-        [DataMember(Order = 11)]
-        public readonly SymbolKey? NavigationSymbolId;
-        [DataMember(Order = 12)]
-        public readonly int NavigationSymbolIndex;
+        public readonly SymbolItemLocation? Location;
```

Comment:
wondering if it is better to always have a non-nullable Location, but in the case of missing references it just has neither indoc nor otherdoc info.  but likely doesn't matter

Reply:
I'd prefer not:
1. so we catch mistakes as early as possible.  having this be nullable means we'll crash when retrieving it if is null.
2. so we don't have to another valid state for the SymbolItemLocation type.  Right now it represents in-doc or other-doc.  This would now mean it could also mean: no location.  But i'd prefer not to do that.